packages/flutter/lib/src/painting/superellipse_shape.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'basic_types.dart';
+import 'border_radius.dart';
+import 'borders.dart';
+import 'edge_insets.dart';
+
+/// Creates a shape constituting a transition between a rectangle and a circle.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Material(
+///     shape: SuperellipseShape(25.0),
+///   ); // Material
```

Comment:
nit: remove the comment `// Material` - IDEs will end up adding a duplicative "ghost" comment if people copy and paste this.